clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -793,19 +812,22 @@ invariant_checker_t::check_for_pc_discontinuity(
                     // Indirect branches we cannot check.
                     !type_is_instr_direct_branch(shard->prev_instr_.instr.type) ||
                     // Conditional fall-through hits the regular case above.
-                    !have_cond_branch_target || memref.instr.addr == cond_branch_target;
+                    !have_cond_branch_target || current_memref_addr == cond_branch_target;
 
                 if (!valid_branch_flow) {
                     error_msg = "Direct branch does not go to the correct target";
                 }
             } else if (cur_instr_decoded != nullptr &&
                        shard->prev_instr_decoded_ != nullptr &&
                        instr_is_syscall(cur_instr_decoded->data) &&
-                       memref.instr.addr == prev_instr_trace_pc &&
+                       current_memref_addr == prev_instr_trace_pc &&
```

Comment:
For signals, we should be comparing `last_instr_in_cur_context_.instr.addr` (not prev_instr_.instr.addr) with current_memref_addr.